client/src/elm/Pages/Prenatal/ProgressReport/View.elm

```diff
@@ -1227,6 +1208,62 @@ viewObstetricalDiagnosisPane language currentDate isChw globalLmpValue firstNurs
         ]
 
 
+viewMedicationHistoryPane : Language -> NominalDate -> Bool -> AssembledData -> Html any
+viewMedicationHistoryPane language currentDate isChw assembled =
+    let
+        allNurseEncountersData =
+            generateAllNurseEncountersData isChw assembled
+                |> List.sortWith (sortByDateDesc .startDate)
+
+        resolveMedicationAdministrationDate measurementFunc =
+            List.map (.measurements >> measurementFunc) allNurseEncountersData
+                |> List.filterMap
+                    (Maybe.andThen
+                        (\( _, measurement ) ->
+                            if measurement.value == AdministeredToday then
+                                Just measurement.dateMeasured
+
+                            else
+                                Nothing
+                        )
+                    )
+                |> List.head
+
+        entriesHeading =
+            div [ class "heading medication" ]
+                [ div [ class "name" ] [ text <| translate language Translate.Medication ]
+                , div [ class "date" ] [ text <| translate language Translate.DateReceived ]
+                ]
+
+        viewEntry medicationType administrationDate =
+            let
+                dateForView =
+                    Maybe.map formatDDMMYYYY administrationDate
+                        |> Maybe.withDefault "--"
+            in
+            div [ class "entry medication" ]
+                [ div [ class "cell name" ] [ text <| translate language <| Translate.MedicationDistributionSign medicationType ]
+                , div [ class "cell date" ] [ p [] [ text dateForView ] ]
+                ]
+    in
+    div [ class "medication-history" ] <|
+        [ viewItemHeading language Translate.MedicationHistory "blue"
+        , div [ class "pane-content" ]
+            [ entriesHeading
+            , resolveMedicationAdministrationDate .calcium
+                |> viewEntry Calcium
```

Comment:
Is it alphabetic order or importance order?

Reply:
There was no special request on this. So, 4 first were alphabetic and then came up a request to add 5-th one, which I added as last.